x-pack/solutions/observability/plugins/observability_onboarding/test/scout/ui/tests/onboarding_ui_validation.spec.ts

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { expect } from '@kbn/scout-oblt';
+import { test } from '../fixtures';
+
+const TEST_TIMEOUT = 3 * 60 * 1000;
+
+test.describe('Onboarding UI Validation', () => {
+  test.describe.configure({
+    timeout: TEST_TIMEOUT,
+    retries: 2,
+  });
+
+  test.beforeEach(async ({ pageObjects, browserAuth }) => {
+    await browserAuth.loginAsAdmin();
+    await pageObjects.onboarding.goto();
+    await pageObjects.onboarding.waitForMainTilesToLoad();
+  });
+
+  test(
+    'should always show Host, Kubernetes, and Cloud tiles',
+    { tag: ['@ess', '@svlOblt'] },
+    async ({ pageObjects }) => {
+      await expect(pageObjects.onboarding.hostUseCaseTile).toBeVisible();
+      await expect(pageObjects.onboarding.kubernetesUseCaseTile).toBeVisible();
+      await expect(pageObjects.onboarding.cloudUseCaseTile).toBeVisible();
+    }
+  );
+
+  test(
+    'should show Application tile and complete mode copy when in complete deployment',
+    { tag: ['@ess', '@svlOblt'] },
+    async ({ pageObjects }) => {
+      test.skip(
+        !(await pageObjects.onboarding.applicationUseCaseTile.isVisible()),
+        'Application tile not visible - likely logs-essentials mode'
+      );
+
+      const hostDescription = await pageObjects.onboarding.getTileDescription('host');
+      const kubernetesDescription = await pageObjects.onboarding.getTileDescription('kubernetes');
+
+      expect(hostDescription).toContain('Monitor your host');
+      expect(kubernetesDescription).toContain('metrics');
+      expect(kubernetesDescription).toContain('traces');
```

Comment:
Copy checks feel a bit too granular for UI tests, they are heavy on the CI and better be used for more high-level stuff: page layout, navigation, cross-component communication, etc.

I'd move copy checks to Jest and test them on a component level using React Testing Library, this would be much faster and more isolated. [Product tier service](https://github.com/elastic/kibana/pull/221281) already has a mock package to test components for different product tiers.

WDYT?